internal/config/config.go

```diff
@@ -198,7 +203,11 @@ func (st ScanType) GetGVKs(d Discovery, log logr.Logger) ([]schema.GroupVersionK
 					// try finding this resource type in another version of this group.
 					continue
 				}
-				gvks = append(gvks, gvk)
+				preferredVersion, err := d.findGroupPreferredVersion(group)
```

Comment:
I'm wondering whether we could refactor this function a bit anyways.

`versionsForGroup` always returns the preferred version as the first element, maybe we could do something with that.

But I couldn't figure out something just by staring at it 🤔

Reply:
I know what you mean. What was the motivation for always returning the preferredVersion first when building a versions array? There's a comment in versionsForGroup explaining _that_ we want to do this, but it doesn't explain why.

Reply:
Uhm the reasoning was to only have one call / functions instead of two. Also it made the code a bit simpler when we loop over it, as we don't need a special case for the preferredversion anymore.

Reply:
Sorry, what I'm trying to ask is why we needed the preferredVersion to be first in that particular array. If we don't (anymore), I'm inclined to remove that ordering code. If we do, we _could_ inject this preferred version instead, or alternatively call versionsForGroup unconditionally just to be able to pluck the preferredVersion from the front of the array...

But I'm not sure if that would actually make any of this more understandable? If you're coming from a performance PoV, I'm not sure it matters a lot - we're looping over a data structure twice instead of once, but only at startup rather than on every reconciliation.